syntax = "proto3";

package NDTS.NTabasco;

service TTabascoGRPC {
    rpc GetScripts (TGetScriptsRequest) returns (TGetScriptsResponse) {}
    rpc GetBatch (TGetBatchRequest) returns (TGetBatchResponse) {}
}

message TGetBatchRequest {
    uint64 taskId = 1;
    uint64 batchIndex = 2;
}

message TGetBatchResponse {
    repeated bytes input = 1;
    repeated bytes output = 2;
}

message TGetScriptsRequest {
    uint64 taskId = 1;
    uint64 buildId = 2;
}

message TGetScriptsResponse {
    bytes build = 1;
    bytes execute = 2;
    uint64 batchCount = 3;
}